avi/resource_avi_cloud.go

```diff
@@ -236,10 +243,104 @@ func ResourceAviCloudRead(d *schema.ResourceData, meta interface{}) error {
 	return err
 }
 
+// Verify cloudState is valid vcenter state to configure mgmt IP
+func isValidCloudState(cloudState string) bool {
+	switch cloudState {
+	case
+		"VCENTER_DISCOVERY_COMPLETE_NO_MGMT_NW",
+		"VCENTER_DISCOVERY_COMPLETE",
+		"VCENTER_DISCOVERY_WAITING_DC",
+		"VCENTER_DISCOVERY_ONGOING":
+		return true
+	}
+	return false
+}
+
+// Wait untill vcenter inventory gets completed to configure mgmt IP
+func waitForVcenterState(cloudUuid string, client *clients.AviClient, maxRetry int) error {
+	var robj interface{}
+	var err error
+	path := "api/vimgrvcenterruntime?cloud_uuid=" + cloudUuid
+	i := 0
+	for ; i < maxRetry; i++ {
+		if err = client.AviSession.Get(path, &robj); err == nil {
+			objCount := robj.(map[string]interface{})["count"].(float64)
+			if objCount == float64(1) {
+				inventoryState := robj.(map[string]interface{})["results"].([]interface{})[0].(map[string]interface{})["inventory_state"].(string)
+				if isValidCloudState(inventoryState) {
+					log.Printf("Got expected inventory state %s", inventoryState)
+					break
+				} else {
+					log.Printf("Didn't get expected inventory state. Current state is %s", inventoryState)
+				}
+			} else {
+				log.Printf("Inventory is not completed")
```

Comment:
we should print the inventory API call error instead of saying inventory is not complete.

Reply:
In the next else we have print the API call error